de.metas.vertical.pharma/src/test/java/de/metas/vertical/pharma/PharmaOrderLineInputValidatorTest.java

```diff
@@ -0,0 +1,265 @@
+package de.metas.vertical.pharma;
+
+import de.metas.ShutdownListener;
+import de.metas.StartupListener;
+import de.metas.order.OrderLineRepository;
+import de.metas.vertical.pharma.model.I_C_BPartner;
+import de.metas.vertical.pharma.model.I_M_Product;
+import de.metas.vertical.pharma.model.interceptor.C_OrderLine;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.test.AdempiereTestHelper;
+import org.compiere.Adempiere;
+import org.compiere.model.*;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.save;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {
+		//		 needed to register the spring context with the Adempiere main class
+		StartupListener.class, ShutdownListener.class,
+
+		// needed so that the spring context can discover those components. Note that there are other ways too, but this one is very fast
+		PharmaBPartnerRepository.class,
+		PharmaOrderLineInputValidator.class,
+		OrderLineRepository.class,
+		PharmaProductRepository.class,
+		C_OrderLine.class,
+})
+public class PharmaOrderLineInputValidatorTest
+{
+	@Rule public ExpectedException thrown = ExpectedException.none();
+
+	private C_OrderLine orderLineInterceptor;
+
+	@BeforeClass
```

Comment:
note: for new unit tests it makes sense to use jupiter. but no need to change this one now